parmed/amber/asciicrd.py

```diff
@@ -61,7 +62,14 @@ def __init__(self, fname, natom, hasbox, mode='r', title=None):
             self._writebox = False
         else:
             raise ValueError("%s mode must be 'r' or 'w'" % type(self).__name__)
-        self._file = genopen(fname, mode)
+        if isinstance(fname, string_types):
+            self._file = genopen(fname, mode)
+            self._own_handle = True
+        elif hasattr(fname, 'read'):
+            self._file = fname
+            self._own_handle = False
+        else:
+            raise RuntimeError("Don't know how to handle %s" % fname)
```

Comment:
`RuntimeError` -> `TypeError`, and then say it's an unsupported type in the error message